airbyte-webapp/src/hooks/services/Feature/FeatureService.tsx

```diff
@@ -44,3 +56,19 @@ export const WithFeature: React.FC<{ featureId: FeatureItem }> = ({
   const { hasFeature } = useFeatureService();
   return hasFeature(featureId) ? <>{children}</> : null;
 };
+
+export const useFeatureRegisterValues = (props?: Feature[] | null): void => {
+  const { registerFeature, unregisterFeature } = useFeatureService();
+
+  useEffect(() => {
+    if (props) {
+      registerFeature(props);
+
+      return () =>
+        unregisterFeature(props.map((feature: Feature) => feature.id));
+    }
+
+    return;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [props]);
+};
```

Comment:
This hook currently does not work properly (and introduces a minor memory leak). `registerFeature` and `unregisterFeature` will be changing every time a feature is added or removed. But this effect hook will keep a reference to the previous `unregisterFeature` in it's cleanup method. That previous `unregisterFeature` itself holds a reference to the `additionFeature` array, that will now also the old one, i.e. when this hook cleans up it basically resets every other feature registration that has happened since it was setup. This is currently not noticeable in the app, since we only use it in one place, but we should nevertheless try to create this in a proper working way, since there'd be a high danger this would otherwise be overlooked when we try using this hook in multiple places.

A simple react component that exposes that broken behavior:

```tsx
import {
  FeatureItem,
  useFeatureRegisterValues,
  useFeatureService,
} from "hooks/services/Feature";
import { useMemo, useState } from "react";

const FeatureToggle: React.FC<{ feature: FeatureItem }> = (props) => {
  const [isEnabled, setIsEnabled] = useState(false);
  // When isEnabled pass `useFeatureRegisterValues` the array with the feature.
  // If the feature is disabled use `null` to clean this feature again (and due to
  // the issue also more features from other components).
  const featureArray = useMemo(() => {
    return isEnabled ? [{ id: props.feature }] : null;
  }, [props.feature, isEnabled]);

  useFeatureRegisterValues(featureArray);
  return (
    <div>
      <button onClick={() => setIsEnabled((prev) => !prev)}>
        Toggle {props.feature} (currently: {String(isEnabled)})
      </button>
    </div>
  );
};

export const TestPage: React.FC = () => {
  const { features } = useFeatureService();

  return (
    <div>
      <code>
        <pre>{JSON.stringify(features, null, 2)}</pre>
      </code>

      <FeatureToggle feature={FeatureItem.TEST1}></FeatureToggle>
      <FeatureToggle feature={FeatureItem.TEST2}></FeatureToggle>
      <FeatureToggle feature={FeatureItem.TEST3}></FeatureToggle>
    </div>
  );
};
```

Let's also please add some unit tests for new hooks like that, that will help test and verify the behavior easier.